codec/codec.go

```diff
@@ -0,0 +1,46 @@
+package codec
```

Comment:
Let's decide whether we take the string match approach or something else first. And then come back to this because a major reason for me to do this is that I do not want to pass content-type string out of the transport package just for extracting the codec info from it in the grpc package.